### ISSUE TITLE
Failing unit tests with float - unit test error checking issue

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos-kernels#322]
This was found testing on White with the OpenMP backend, float scalar types, and using the module `devpack/20180521/openmpi/3.1.0/gcc/7.2.0/cuda/9.2.88` with openblas swapped for `netlib/3.8.0/gcc/7.2.0`.

Following PR #320 to fix the compilation error reported in #316 when enabling floats as scalar types, the following tests need to be fixed for error checking:

- [x] abs_float
- [x] abs_mv_float
- [x] nrm2_squared_float
- [x] nrm2_squared_mv_float
- [x] reciprocal_float
- [x] reciprocal_mv_float
- [x] scal_mv_float
- [x] team_scal_mv_float
- [x] scal_float
- [x] team_scal_float
- [x] sparse_trsv_mv_kokkos_complex_float

@kyungjoo-kim helped with fixing the error checking on the first couple tests, hoping to pattern match for the rest.

**Comment by ndellingwood** [comment]
Also seeing some linking errors if I only build with `float` types, rather than `float,double`, e.g.

```
Test_OpenMP_Graph_graph_color_d2.o: In function
`KokkosGraph::Impl::GraphColorD2_MatrixSquared<KokkosKernels::Experimental::KokkosKernelsHandle<int const, int, double, Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<int*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >, Kokkos::View<int*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> > >::color_graph_d2_matrix_squared()':

/ascldap/users/ndellin/kokkos-kernels/testing/White-Cuda9.2-GCC7.3-float/install/include/KokkosSparse_spgemm_symbolic.hpp:183:
undefined reference to `KokkosSparse::Impl::SPGEMM_SYMBOLIC<KokkosKernels::Experimental::KokkosKernelsHandle<int const, int const, double const, Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<int*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, false, false>::spgemm_symbolic(KokkosKernels::Experimental::KokkosKernelsHandle<int const, int const, double const, Kokkos::OpenMP, Kokkos::HostSpace, Kokkos::HostSpace>*, int, int, int, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, bool, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<int const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >, bool, Kokkos::View<int*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<1u> >)'
```

**Comment by ndellingwood** [comment]
Getting a PR ready for this, thanks to @kyungjoo-kim for help!

**Comment by ndellingwood** [comment]
PR #322 issued with fixes for the above.

**Comment by ndellingwood** [comment]
Cross-reference older issue #97.

**Comment by ndellingwood** [comment]
PR updated with fixes detected by gcc/7.2, results of that compiler check on White with OpenMP:

```
bash-4.2$ ../../scripts/test_all_sandia gcc/7.2 --build-list=OpenMP --skip-hwloc
Running on machine: white
Going to test compilers:  gcc/7.2.0
Testing compiler gcc/7.2.0
  Starting job gcc-7.2.0-OpenMP-release
  PASSED gcc-7.2.0-OpenMP-release
#######################################################
PASSED TESTS
#######################################################
gcc-7.2.0-OpenMP-release build_time=235 run_time=225
#######################################################
FAILED TESTS
#######################################################
```
